amazon/ion/ioncmodule.c

```diff
@@ -557,16 +559,20 @@ iERR ionc_write_value(hWRITER writer, PyObject* obj, PyObject* tuple_as_sexp) {
         }
 
         ION_TIMESTAMP timestamp_value;
-        PyObject *fractional_seconds, *fractional_decimal_tuple, *py_exponent, *py_digits;
+        PyObject *fractional_seconds, *fractional_decimal_tuple, *py_exponent, *py_digits, *precision_attr;
         int year, month, day, hour, minute, second;
         short precision, fractional_precision;
         int final_fractional_precision, final_fractional_seconds;
-        if (PyObject_HasAttrString(obj, "precision") && PyObject_GetAttrString(obj, "precision") != Py_None) {
+        precision_attr = PyObject_GetAttrString(obj, "precision");
+        if (precision_attr == NULL) {
```

Comment:
Do we have a test that tries to write a std `datetime`? I may misunderstand which branches that would hit or the indentation, but I think that would fail with this change.

Reply:
> Do we have a test that tries to write a std `datetime`? I may misunderstand which branches that would hit or the indentation, but I think that would fail with this change.

This change did fail the build process. I'll look into the issue and fix the failure. We have the test for timestamp(test_timestamp.py) which has passed locally. I might need to look into the test file to see if we have the test for writing std `datetime` and figure out the reason why it's failed in CI/CD workflow.

Reply:
There is test in [test_roundtrip](https://github.com/amazon-ion/ion-python/blob/c9907633d973918cd1c26a2b710a12ce67db23ba/tests/test_simpleion.py#L596), one of the parameters is [datetime](https://github.com/amazon-ion/ion-python/blob/c9907633d973918cd1c26a2b710a12ce67db23ba/tests/test_simpleion.py#L517). This test was failed because of this change, after updating in commit [4f4788d](https://github.com/amazon-ion/ion-python/pull/319/commits/4f4788d4ccde0a6956233793129826ffb9cfb5d8), this test is passed.